src/clima_sensor_us.rs

```diff
@@ -0,0 +1,560 @@
+				.expect("Error occurred while setting RTS ti RTS-UP");
+			conn.rtu_set_custom_rts(RequestToSendMode::RtuRtsUp)
+				.expect("Error occurred while setting custom RTS-function");
+
+			conn.connect()
+				.expect("Error occurred while connecting to Clima-Sensor");
+		}
+	}
+
+	/// This function should be called periodically to check the sensors' values.
+	/// if temp > 30°C and no wind, then a warning should be issued
+	/// if temp > 35°C a warning should be issued
+	/// if temp < 20° either warning is removed
+	/// (Input Register - 0x04) temp-reg address 0x76C1; typ S32; real_result = response_temp/10
+	/// (Input Register - 0x04) wind-reg address 0x7533; typ U32; real_result = response_wind/10
+	/// The return value is bool on success, true if alarm is active and false is alarm is not active
+	/// If no ctx is configured the this function returns always false, so no warning is triggered
+	pub fn handle(&mut self) -> Result<TempWarningStateChange, std::io::Error> {
+		match &self.ctx {
+			Some(conn) => {
+				let mut response_temp = vec![0u16; 2];
+				let mut response_wind = vec![0u16; 2];
+
+				if let Err(error) = conn.read_input_registers(REG_AIR_TEMP, 2, &mut response_temp) {
+					return Err(std::io::Error::new(
+						std::io::ErrorKind::Other,
+						error.to_string(),
+					));
+				}
+				if let Err(error) =
+					conn.read_input_registers(REG_MEAN_WIND_SPEED, 2, &mut response_wind)
+				{
+					return Err(std::io::Error::new(
+						std::io::ErrorKind::Other,
+						error.to_string(),
+					));
+				}
+
+				let temp: f32 = (conv_vec_to_value_s(response_temp).unwrap() as f32) / 10.0;
+				let wind: f32 = (conv_vec_to_value_u(response_wind).unwrap() as f32) / 10.0;
+				#[cfg(debug_assertions)]
+				println!(
+					"Weatherstation: temperature {} °C, windspeed {} m/s",
+					temp, wind
+				);
+				//check if new data should be published to opensensemap.org
+				if self.opensensemap_counter == UPDATE_FREQUENCY {
+					self.opensensemap_counter = 0;
+					match self.publish_to_opensensemap() {
+						Ok(_) => {}
+						Err(error) => return Err(error),
+					}
+				} else {
+					self.opensensemap_counter += 1;
+				}
+
+				Ok(self.set_warning_active(temp, wind))
+			}
+			None => Ok(TempWarningStateChange::None),
+		}
+	}
+
+	/// This function is used to set the warning_active varibale and compare it with the new value.
+	fn set_warning_active(&mut self, temp: f32, wind: f32) -> TempWarningStateChange {
+		let new_warning: TempWarning;
+		let mut result: TempWarningStateChange = TempWarningStateChange::None;
+
+		if temp > 35.0 {
+			new_warning = TempWarning::WarningTemp;
+		} else if temp > 30.0
+			&& wind < 0.3
+			&& !matches!(self.warning_active, TempWarning::WarningTemp)
+		{
+			new_warning = TempWarning::WarningTempNoWind;
+		} else if temp > 23.0
+			&& !matches!(self.warning_active, TempWarning::WarningTemp)
+			&& !matches!(self.warning_active, TempWarning::WarningTempNoWind)
```

Comment:
```suggestion
			&& !matches!(self.warning_active, TempWarning::WarningTemp | TempWarning::WarningTempNoWind)
```

you can match on multiple variants here

Reply:
Thanks for that information, looks much cleaner. :smiley: